### DISCUSSION TITLE
Spinner Control syntax for usage

### DISCUSSION THREAD

**Discussion by westerdaled** [microsoft/powercat-creator-kit#61] [Q&A] [answered]
I did have  go at trying to use this earlier after I dragged the control on to a screen. 

I am guess this syntax is not the way to use this control.   

```
spnCallStoredProcedure.Visible =true;
spnCallStoredProcedure.IsLoading = true;
ClearCollect(colDatasources, 'PowerApp->GetSomeThing.Run().ResultSets.Table1.Value);
spnCallStoredProcedure.IsLoading = false;
spnCallStoredProcedure.Visible =false;
```

Should I refactor the call to use this syntax - just don't know how this references the above control?

```
"Spinner (0.0.18)'":

    IsLoading: true    
    Visible: true
```

**Comment by Ramakrishnan24689** [comment, marked as answer]
Hi @westerdaled,

Thanks for reaching out.
From the syntax, I am assuming that you want to make use of the spinner control which needs to appear until a PowerAutomate has completed its run and need to know the syntax to be used.

Here is an example - sample code:

`UpdateContext({ var_showLoader: true });
/*Some code - PowerAutomate Run for example*/
ClearCollect(colDatasources, 'PowerApp->GetSomeThing.Run().ResultSets.Table1.Value);
UpdateContext({ var_showLoader: false });`

Where '**var_showLoader**' will define whether Spinner should appear or not. This value need to be passed to **IsLoading** Property.

More on this here - https://docs.microsoft.com/en-us/power-platform/guidance/creator-kit/spinner#configuring-spinner-loading-behavior

Hope this helps.

**Comment by westerdaled** [comment]
Ah thanks  for the replying.  I think when I last built a simple component , I set properties in the way I previously described.  Hence the confusion. Thanks again.
